src/Main.elm

```diff
@@ -28,25 +32,109 @@ init =
     { display = "0"
     , displayStatus = ShowingResults
     , decimalStatus = NoDecimal
+    , valueA = 0
+    , actionA = Blank
+    , valueB = 0
+    , actionB = Blank
     }
 
 
+type DisplayStatus
+    = ShowingResults
+    | NotResults
+
+
+type DecimalStatus
+    = YesDecimal
+    | NoDecimal
+
+
+orderOfOperations : Model -> Model
+orderOfOperations model =
+    case ( model.actionA, model.actionB ) of
+        ( Add, Equals ) ->
+            simpleMath model
+
+        ( Add, Add ) ->
+            simpleMath model
+
+        ( _, _ ) ->
+            model
+
+
+simpleMath : Model -> Model
+simpleMath model =
+    case model.actionA of
+        Add ->
+            { model
+                | display = String.fromFloat <| model.valueA + model.valueB
+                , valueA = 0
+                , actionA = Blank
+                , valueB = 0
+                , actionB = Blank
+                , displayStatus = ShowingResults
+            }
+
+        _ ->
+            model
+
+
 
 -- Update
 
 
 type Msg
     = UpdateDisplay CalButton
+    | Saveinput CalButton
 
 
-type DisplayStatus
-    = ShowingResults
-    | NotResults
+buttonToAction : CalButton -> Action
+buttonToAction button =
+    case button of
+        CalAdd ->
+            Add
 
+        CalEqual ->
+            Equals
+
+        _ ->
+            Blank
+
+
+calActionHit : Model -> CalButton -> Model
+calActionHit model button =
+    orderOfOperations <| saveAction model button
 
-type DecimalStatus
-    = YesDecimal
-    | NoDecimal
+
+saveAction : Model -> CalButton -> Model
+saveAction model button =
+    case ( model.actionA, model.actionB ) of
+        ( Blank, Blank ) ->
+            { model
+                | valueA = displatToFloat model.display
+                , actionA = buttonToAction button
+                , displayStatus = ShowingResults
+            }
+
+        ( _, Blank ) ->
+            { model
+                | valueB = displatToFloat model.display
+                , actionB = buttonToAction button
+                , displayStatus = ShowingResults
+            }
+
+        ( _, _ ) ->
+            model
+
+
+displatToFloat : String -> Float
+displatToFloat string =
+    case String.toFloat string of
+        Just float ->
+            float
+
+        Nothing ->
+            0
```

Comment:
:cry: This makes me sad. Maybe the calculator should keep numbers internally to avoid having to parse? That may need to be refactored in another PR.